CP-18272 - add integration tests

lindig opened the pull request:
This PR adds tests that validate the binary format of the RRD files written by the library. The files are read using a tool built with the rrd-transport library that is the ultimate consumer of these files. The test cases cover the case that files are updated incrementally.

Tests broke earlier because there is currently a change in the RRD transport format that is not fully released. To deal with it, this commit introduces an `#ifdef` section. It will have to be changed, once the changes are fully released.

lindig committed:
CP-18272 introduce #ifdef around change in RRD format change 1.0.0/1.1.0

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

lindig committed:
CP-18272 add integration tests using ocaml-rrd-transport framework

This commit adds test that use rrdreader built with the OCaml
rrd-transport framework. Files written by the library are read back the
by the tool. This is an integration test as the rrd-transport framework
is the ultimate consumer of RRD files.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

lindig committed:
CP-18272 change URL in README to point to correct Travis CI icon

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

lindig committed:
CP-18271 fix indentation

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

euanh commented on ocaml/rrdreader.ml:
```diff
@@ -25,6 +25,14 @@
 
```
I think you should move this utility into the rrd-transport package and install the binary from there.   You need to depend on rrd-transport anyway, and if you do this you can remove the OCaml dependency from this library.   The reader utility may also be useful to other users of rrd-transport.

lindig committed:
CP-18272 include Parson library as directory (was: Git submodule)

Including the code makes packaging easier. The MIT license of Parson is
compatible with the MIT license of rrd-client-lib. The name of the
parson directory points to the commit we are using.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

lindig committed:
CP-18272 change signature of rrd_sample to support testing

The commit changes the signature of rrd_sample to

    int rrd_sample(RRD_PLUGIN * plugin, time_t (*t)(time_t*));

The caller can provide a function to obtain a time stamp or provide NULL
(the typical case). This permits to control the time stamps in RRD files
for debugging and testing.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

lindig committed:
CP-18272 split off integration tests into new target test-integration

We would like to split tests into those that do not require many
external dependencies (test) and those that do (test-integrate).

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

lindig committed:
CP-18271 remove OCaml code, use external rrdreader tool for testing

The rrdreader tool has moved to the ocaml-rrd-transport 1.1.1 package on
which we rely on anyway. This allows us to remove all OCaml code in this
repository.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>
